build_test.go

```diff
@@ -0,0 +1,185 @@
+package composer_test
+
+import (
+	"bytes"
+	"github.com/paketo-buildpacks/composer"
+	"github.com/paketo-buildpacks/composer/fakes"
+	"github.com/paketo-buildpacks/packit/v2"
+	"github.com/paketo-buildpacks/packit/v2/chronos"
+	"github.com/paketo-buildpacks/packit/v2/fs"
+	"github.com/paketo-buildpacks/packit/v2/postal"
+	"github.com/sclevine/spec"
+	"os"
+	"path/filepath"
+	"testing"
+	"time"
+
+	. "github.com/onsi/gomega"
+)
+
+func testBuild(t *testing.T, context spec.G, it spec.S) {
+	var (
+		Expect = NewWithT(t).Expect
+
+		cnbDir     string
+		workingDir string
+		layersDir  string
+
+		clock     chronos.Clock
+		timeStamp time.Time
+
+		buffer            *bytes.Buffer
+		dependencyManager *fakes.DependencyManager
+		entryResolver     *fakes.EntryResolver
+
+		build packit.BuildFunc
+	)
+
+	it.Before(func() {
+		buffer = bytes.NewBuffer(nil)
+		logEmitter := composer.NewLogEmitter(buffer)
+
+		var err error
+		layersDir, err = os.MkdirTemp("", "layers")
+		Expect(err).NotTo(HaveOccurred())
+
+		workingDir, err = os.MkdirTemp("", "working-dir")
+		Expect(err).NotTo(HaveOccurred())
+
+		cnbDir, err = os.MkdirTemp("", "cnb")
+		Expect(err).NotTo(HaveOccurred())
+
+		timeStamp = time.Now()
+		clock = chronos.NewClock(func() time.Time {
+			return timeStamp
+		})
+
+		dependencyManager = &fakes.DependencyManager{}
+		entryResolver = &fakes.EntryResolver{}
+
+		build = composer.Build(logEmitter, dependencyManager, entryResolver, clock)
+	})
+
+	it.After(func() {
+		Expect(os.RemoveAll(layersDir)).To(Succeed())
+		Expect(os.RemoveAll(workingDir)).To(Succeed())
+		Expect(os.RemoveAll(cnbDir)).To(Succeed())
+	})
+
+	it("returns a result that installs composer", func() {
+		entryResolver.ResolveCall.Returns.BuildpackPlanEntry = packit.BuildpackPlanEntry{
+			Name: "composer",
+			Metadata: map[string]interface{}{
+				"version": "resolved-composer-dependency",
+			},
+		}
+
+		entryResolver.MergeLayerTypesCall.Returns.Launch = true
+		entryResolver.MergeLayerTypesCall.Returns.Build = true
```

Comment:
I'd expect to see fake set-up calls in an `it.Before` since they're less about what we're testing and more about setting up the environment. I'd also expect to see a test where these are not set to true, so we know that the build result has the correct layer metadata set.